packages/expo-print/ios/ExpoPrintWithPrinter.swift

```diff
@@ -11,15 +11,16 @@ public class ExpoPrintWithPrinter {
 
   func startPrint(options: PrintOptions, promise: Promise) {
     if let uri = options.uri {
-      guard let printingData = dataFromUri(uri: uri) else {
-        promise.reject(InvalidUrlException())
-        return
+      dataFromUri(uri: uri) { [self] data in
+        if let printingData = data {
+          self.printWithData(printingData: printingData, options: options, promise: promise)
```

Comment:
We should use weak reference in here: 
```suggestion
      dataFromUri(uri: uri) { [weak self] data in
        if let self = self, let printingData = data {
          self.printWithData(printingData: printingData, options: options, promise: promise)
        ```